tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -959,31 +959,29 @@ def test__delete_dataset(self):
             (None, "none", 401),
             (self.make_not_owner_header, "not_owner", 403),
         ]
+
+        def _delete(collection_id, dataset_id):
+            test_url = f"/curation/v1/collections/{collection_id}/datasets/{dataset_id}"
+            headers = auth() if callable(auth) else auth
+            return self.app.delete(test_url, headers=headers)
+
         for auth, auth_description, expected_status_code in auth_credentials:
```

Comment:
```suggestion
    for auth, auth_description, expected_status_code in auth_credentials:
        dataset = self.generate_dataset(
            statuses=[DatasetStatusUpdate(DatasetStatusKey.UPLOAD, DatasetUploadStatus.UPLOADING)],
            publish=False,
        )
        tests = [
            ("with version ids", dataset.collection_id, dataset.dataset_id), 
            ("with canonical ids",dataset.collection_version_id,dataset.dataset_version_id)
        ]
        for description, collection_id, dataset_id in tests:
            with self.subTest(f"{auth_description} {expected_status_code} {description}"):
                # with version ids

                response = _delete(dataset.collection_version_id, dataset.dataset_version_id)
                self.assertEqual(expected_status_code, response.status_code)
```

Reply:
I don't think that's gonna work because the dataset needs to be generated before each deletion.

Reply:
ah duh. Then I'd make this into two separate tests.